sources/api/apiclient/src/get.rs

```diff
@@ -24,13 +29,79 @@ where
     }
 
     // Merge results together.
-    results
+    let mut merged = results
         .into_iter()
         .reduce(|mut merge_into, merge_from| {
             merge_json(&mut merge_into, merge_from);
             merge_into
         })
-        .context(error::NoPrefixesSnafu)
+        .context(error::NoPrefixesSnafu)?;
+
+    // Remove excluded prefixes
+    if !exclude.is_empty() {
+        remove_prefixes(&mut merged, &exclude)?;
+    }
+
+    Ok(merged)
+}
+
+/// Removes keys matching any of the given prefixes from the JSON value.
+/// Uses dotted key notation (e.g., "settings.network" matches "settings.network.hostname").
+/// Prefixes without trailing dots match any key starting with that prefix.
+/// Prefixes with trailing dots only match keys with that exact prefix followed by more path segments.
+fn remove_prefixes(value: &mut serde_json::Value, prefixes: &[String]) -> Result<()> {
+    remove_prefixes_recursive(value, "", prefixes);
+    Ok(())
+}
+
+/// Recursively removes keys from JSON, tracking the dotted path as we traverse.
+/// Returns the number of keys remaining in the object after removal.
+fn remove_prefixes_recursive(
+    value: &mut serde_json::Value,
+    parent_path: &str,
+    prefixes: &[String],
+) -> usize {
+    let serde_json::Value::Object(map) = value else {
+        // Non-object values count as 1 (they exist)
+        return 1;
+    };
+
+    // Build paths once and determine which keys to remove
+    let mut keys_to_remove = Vec::new();
+    let mut keys_to_recurse = Vec::new();
+
+    for key in map.keys() {
+        let full_path = if parent_path.is_empty() {
+            key.to_string()
+        } else {
+            format!("{}.{}", parent_path, key)
+        };
+
+        let should_remove = prefixes.iter().any(|prefix| full_path.starts_with(prefix));
+
+        if should_remove {
+            keys_to_remove.push(key.clone());
+        } else {
+            keys_to_recurse.push((key.clone(), full_path));
+        }
+    }
+
+    // Remove matching keys
+    for key in keys_to_remove {
+        map.remove(&key);
+    }
+
+    // Recursively process remaining nested objects and remove if empty
+    for (key, full_path) in keys_to_recurse {
+        if let Some(nested_value) = map.get_mut(&key) {
+            let remaining = remove_prefixes_recursive(nested_value, &full_path, prefixes);
+            if remaining == 0 {
+                map.remove(&key);
+            }
+        }
+    }
```

Comment:
nit: we could make this a one pass that is included in the for-loop above. 

I am also fine with this second pass for better readability. This may be something we need to worry about if the number of settings grows more significantly.

Reply:
Doing this in a later loop helps avoid needing a mutable borrow of `map` while iterating through an immutable borrow. I'd like to keep it for clarity for now.

I agree that we'll be in trouble if we end up with a large number of settings, or lots of excluded prefixes to match against.